packages/mana/manabasevector.go

```diff
@@ -175,6 +164,75 @@ func (m *ManaBaseVector) Book(txInfo *TxInfo) {
 	}
 }
 
+func (m *ManaBaseVector) BookEpoch(created []*ledger.OutputWithMetadata, spent []*ledger.OutputWithMetadata) {
+	var revokeEvents []*RevokedEvent
+	var pledgeEvents []*PledgedEvent
+	var updateEvents []*UpdatedEvent
+	// only lock mana vector while we are working with it
+	func() {
+		m.Lock()
+		defer m.Unlock()
+		// first, revoke mana from previous owners
+		for _, output := range spent {
+			idToRevoke := m.getIDBasedOnManaType(output)
+			outputIOTAs, _ := output.Output().(devnetvm.Output).Balances().Get(devnetvm.ColorIOTA)
+			oldMana := m.getOldManaAndRevoke(idToRevoke, float64(outputIOTAs))
+			// save events for later triggering
+			revokeEvents = append(revokeEvents, &RevokedEvent{idToRevoke, float64(outputIOTAs), output.OutputMetadata().CreationTime(), m.Type(), output.ID().TransactionID, output.ID()})
```

Comment:
Can you spell the fields out of these events one per line? It would be more redeable.